arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -1065,7 +1187,7 @@ fun <E, A, B, E2 : E> IOOf<E, A>.flatMap(f: (A) -> IOOf<E2, B>): IO<E2, B> =
  * }
  * ```
  */
-fun <E, A> IO<E, EitherOf<E, A>>.flattenEither(): IO<E, A> =
+fun <E : Throwable, A> IO<EitherOf<E, A>>.flattenEither(): IO<A> =
```

Comment:
👀 @nomisRev @JorgeCastilloPrz Although not as useful as before, I left this functions here but restraining E to throwable, so they can still be used in certain situations